tasks/tslint.js

```diff
@@ -50,9 +51,14 @@ module.exports = function (grunt) {
                 }
                 options.configuration = configuration;
 
-                var contents = grunt.file.read(filepath);
-                var linter = new Linter(filepath, contents, options);
-                var result = linter.lint();
+                var lintOptions = {
+                    fix: options.fix,
+                    formatter: options.formatter
+                }
+
+                var linter = new Linter.Linter(lintOptions, Linter.Linter.createProgram(configuration.path));
```

Comment:
this doesn't look quite right; this ends up creating a new program for every file in the input glob. a TypeScript "program" should only be created when that option is specified in the TSLint API.